sdk/schema/plugin.go

```diff
@@ -117,3 +118,23 @@ func (p Plugin) DeepValidate() []ValidationReport {
 
 	return reports
 }
+
+func (p Plugin) MarshalJSON() ([]byte, error) {
+	if len(p.Credentials) == 0 {
+		return nil, nil
+	}
+	plugin := struct {
+		PlatformName  string `json:"platform_name"`
+		ManagementURL string `json:"management_url,omitempty"`
+	}{
+		PlatformName: p.Platform.Name,
+		ManagementURL: func() string {
+			if p.Credentials[0].ManagementURL != nil {
+				return p.Credentials[0].ManagementURL.String()
```

Comment:
I noticed that this results in escaping of the & character into its unicode representation. So, the Homebrew management URL becomes https://github.com/settings/tokens/new?scopes=gist,repo,workflow\u0026description=Homebrew and this fails to work on GitHub (missing token name):

<img width="1141" alt="image" src="https://github.com/1Password/shell-plugins/assets/18581859/ade61735-4abd-4369-b5d7-9ef680724864">

Depending on how this management URL is used on the CLI side of the banner implementation, we should make sure to use the raw unescaped string.

Reply:
These are great questions! But I recommend we take them up within the internal MR, as this PR lacks the context for them.